src/test/lib/DataTest.cpp

```diff
@@ -2692,6 +2696,112 @@ QuicTestStreamAbortConnFlowControl(
     TEST_TRUE(Context.ClientStreamShutdownComplete.WaitTimeout(TestWaitTimeout));
 }
 
+struct StreamBlockUnblockConnFlowControl {
+    CxPlatEvent ClientStreamShutdownComplete;
+    CxPlatEvent ClientStreamSendComplete;
+    CxPlatEvent ServerStreamReceive;
+    CxPlatEvent ServerConnectionPeerNeedsStreams;
+    uint32_t NeedsStreamCount {0};
+
+    static QUIC_STATUS ClientStreamCallback(_In_ MsQuicStream*, _In_opt_ void* Context, _Inout_ QUIC_STREAM_EVENT* Event) {
+        auto TestContext = (StreamBlockUnblockConnFlowControl*)Context;
+        if (Event->Type == QUIC_STREAM_EVENT_SEND_COMPLETE && !Event->SEND_COMPLETE.Canceled) {
+            TestContext->ClientStreamSendComplete.Set();
+        }
+        else if (Event->Type == QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE) {
+            TestContext->ClientStreamShutdownComplete.Set();
+        }
+        return QUIC_STATUS_SUCCESS;
+    }
+
+    static QUIC_STATUS ServerStreamCallback(_In_ MsQuicStream*, _In_opt_ void* Context, _Inout_ QUIC_STREAM_EVENT* Event) {
+        auto TestContext = (StreamBlockUnblockConnFlowControl*)Context;
+        if (Event->Type == QUIC_STREAM_EVENT_RECEIVE) {
+            TestContext->ServerStreamReceive.Set();
+        }
+        return QUIC_STATUS_SUCCESS;
+    }
+
+    static QUIC_STATUS ServerConnCallback(_In_ MsQuicConnection* Connection, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT* Event) {
+        auto TestContext = (StreamBlockUnblockConnFlowControl*)Context;
+
+        if (Event->Type == QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED) {
+            new(std::nothrow) MsQuicStream(Event->PEER_STREAM_STARTED.Stream, CleanUpAutoDelete, ServerStreamCallback, Context);
+        } else if (Event->Type == QUIC_CONNECTION_EVENT_PEER_NEEDS_STREAMS) {
+            TestContext->NeedsStreamCount += 1;
+            TestContext->ServerConnectionPeerNeedsStreams.Set();
+            uint16_t count = (uint16_t)(Event->PEER_NEEDS_STREAMS.StreamLimit + 1);
+
+            if(Event->PEER_NEEDS_STREAMS.Bidirectional) {
+                Connection->SetSettings(MsQuicSettings{}.SetPeerBidiStreamCount(count));
+            } else {
+                Connection->SetSettings(MsQuicSettings{}.SetPeerUnidiStreamCount(count));
+            }
+        }
+        return QUIC_STATUS_SUCCESS;
+    }
+
+};
+
+void
+QuicTestStreamBlockUnblockConnFlowControl(
+    _In_ BOOLEAN Bidirectional
+    )
+{
+    MsQuicRegistration Registration(true);
+    TEST_QUIC_SUCCEEDED(Registration.GetInitStatus());
+
+    // Server flow control: UnidirectionalStream : 0, BidirectionalStream : 0
+    MsQuicConfiguration ServerConfiguration(Registration, "MsQuicTest", MsQuicSettings().SetPeerBidiStreamCount(0).SetPeerUnidiStreamCount(0).SetConnFlowControlWindow(200), ServerSelfSignedCredConfig);
```

Comment:
`.SetPeerBidiStreamCount(0).SetPeerUnidiStreamCount(0)` aren't actually necessary, as these are the default values. Let's go ahead and simplify this by removing this.

Reply:
ok